nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/event/pipe/queue/QueueBaseEventBundlePipe.java

```diff
@@ -64,9 +61,8 @@ public void initPipe(String name, Map<String, String> params) {
                 log.error("Unable to read batchSize parameter", e);
             }
         }
-
         queue = new ConcurrentLinkedQueue<>();
-        consumerTPE = new ThreadPoolExecutor(1, 1, 60, TimeUnit.MINUTES, new LinkedBlockingQueue<Runnable>());
+        consumerTPE = new ThreadPoolExecutor(1, 1, 60, TimeUnit.MINUTES, new LinkedBlockingQueue<>());
```

Comment:
Moved the class to tests and take the executor control out of the nested Runnable. 
A bit fixed the shutdown process.